tools/flash_shelly.sh

```diff
@@ -0,0 +1,324 @@
+#  usage: ./flash_shelly.sh -u shelly1-034FFF.local
+
+
+function check_brew {
+  if [ "$(which brew 2>/dev/null)" != "" ]; then
+    return 0
+  fi
+
+  while true; do
+    read -p "brew is not installed, would you like to install it ?" yn
+    case $yn in
+        [Yy]* ) install_brew; break;;
+        [Nn]* ) echo "brew is required for this script to fuction. now exiting..."; exit 1;;
+        * ) echo "Please answer yes or no.";;
+    esac
+  done
+
+  echo -e '\033[1mInstalling brew...\033[0m'
+  echo -e '\033[1mPlease follow instructions...\033[0m'
+  echo -e '\033[1mYou will be asked for your password to install breww...\033[0m'
+  echo -e '\033[1m \033[0m'
+  /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install.sh)"
+}
+
+if [ "$(which timeout 2>/dev/null)" == "" ]; then
+  check_brew
+  echo -e '\033[1mInstalling coreutils...\033[0m'
+  brew install coreutils
+fi
+
+if [ "$(which jq  2>/dev/null)" == "" ]; then
+  check_brew
+  echo -e '\033[1mInstalling jq...\033[0m'
+  brew install jq
+fi
+
+function convert_to_integer {
+  echo "$@" | awk -F "." '{ printf("%03d%03d%03d", $1,$2,$3); }';
+}
+
+function download {
+  curl -L -o shelly-flash.zip "$1"
+}
+
+function write_flash {
+  f_device=$1
+  f_lfw=$2
+  f_durl=$3
+  f_official=$4
+  flashed=false
+  if [ $f_official == false ]; then
+    download "$f_durl"
+    flashcmd="curl -F file=@shelly-flash.zip http://$f_device/update"
+  else
+    flashcmd="curl http://$f_device/ota?url=$f_dlurl"
+  fi
+  if [ -f shelly-flash.zip ] || [ $f_official == "true" ];then
+    while true; do
+      read -p "Are you sure you want to flash $f_device to firmware version $f_lfw ? " yn
+      case $yn in
+        [Yy]* ) echo "Now Flashing.."; $($flashcmd); break;;
+        [Nn]* ) echo "Skipping..";break;;
+        * ) echo "Please answer yes or no.";;
+      esac
+    done
+  fi
+  echo "waiting for $f_device to reboot"
+  sleep 10
+  if [ $f_official == "false" ]; then
+    if [ $(curl -qs -m 5 http://$f_device/rpc/Shelly.GetInfo | jq -r .version) == $f_lfw ];then
+      echo "Successfully flashed $f_device to $f_lfw"
+    else
+      echo "Flash failed!!!"
+    fi
+    rm -f shelly-flash.zip
+  else
+    if [ $(curl -qs -m 5 http://$f_device/Shelly.GetInfo | jq -r .fw | awk '{split($0,a,"/v"); print a[2]}' | awk '{split($0,a,"@"); print a[1]}') == $f_lfw ];then
+      flashed=true
+      echo "Successfully flashed $f_device to $f_lfw"
+    else
+      echo "still waiting for $f_device to reboot"
+      sleep 10
+    fi
+    if [ flashed == true ] || [ $(curl -qs -m 5 http://$f_device/Shelly.GetInfo | jq -r .fw | awk '{split($0,a,"/v"); print a[2]}' | awk '{split($0,a,"@"); print a[1]}') == $f_lfw ];then
+      echo "Successfully flashed $f_device to $f_lfw"
+    else
+      echo "Flash failed!!!"
+    fi
+  fi
+  read -p "Press enter to continue"
+}
+
+function probe_info {
+  official=false
+  flash=null
+  p_device="$1"
+  p_action="$2"
+  p_do_all="$3"
```

Comment:
unused variable

Reply:
which one ? `p_do_all` ? I just gave it a local viable as it is being pass to the function.

Reply:
yes, and it's not used by the function hence no need to pass it.

Reply:
ok.